version: "3.3"

services:

  traefik:
    image: "traefik:v2.6"
    container_name: "traefik"
    command:
     - "--api.insecure=true"
     - "--api.dashboard=true"
     - "--providers.docker"
     - "--log.level=DEBUG"
   labels:
     - "traefik.enable=true"
     - "traefik.http.routers.api.entryPoints=traefik"
     - "traefik.http.routers.api.rule=PathPrefix(`/api`) || PathPrefix(`/dashboard`)"
     - "traefik.http.routers.api.service=api@internal"
     - "traefik.http.routers.api.middlewares=api-auth"
     - "traefik.http.middlewares.api-auth.basicauth.users=test:$$apr1$$H6uskkkW$$IgXLP6ewTrSuBkTrqE8wj/" # test:test
    ports:
      - "80:80"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.rule=Host(`traefik.svc.gweggemans.nl`)"
    networks:
      - traefik

networks:
  traefik:
    external: true